src/mod/asr_tts/mod_unimrcp/mod_unimrcp.c

```diff
@@ -3787,8 +3789,12 @@ static apt_bool_t recog_on_message_receive(mrcp_application_t *application, mrcp
 static apt_bool_t recog_stream_open(mpf_audio_stream_t *stream, mpf_codec_t *codec)
 {
 	speech_channel_t *schannel = (speech_channel_t *) stream->obj;
-	recognizer_data_t *r = (recognizer_data_t *) schannel->data;
 
+	if (!schannel){
+		return FALSE;
+	}
+
+	recognizer_data_t *r = (recognizer_data_t *) schannel->data;
```

Comment:
This will cause a compile issue, you need to declare the variable before the `if(!schannel)`

Reply:
Rushed the changes here, I've made amendment. Thanks!